src/bvals/boundary_conditions.cpp

```diff
@@ -134,66 +98,90 @@ void GenericBC(std::shared_ptr<MeshBlockData<Real>> &rc, bool coarse) {
   const int offset = 2 * ref + (INNER ? -1 : 1);
 
   pmb->par_for_bndry(
-      label, nb, domain, coarse,
+      label, nb, domain, el, coarse,
       KOKKOS_LAMBDA(const int &l, const int &k, const int &j, const int &i) {
         if (!q.IsAllocated(l)) return;
         if (TYPE == BCType::Reflect) {
           const bool reflect = (q.VectorComponent(l) == DIR);
```

Comment:
Sorry this is the wrong place to ask this question and I should have figured this out in a previous MR:

Does the current Parthenon implementation imply that face-centered quantities at the, for example, X1-face, always correspond to a vector X1 component? Or could one equally define a non-vector variable there, or even a three-vector (for example in the three face components of a stress tensor).

Reply:
I'm successfully using 3-component face-centered variables (velocities) for upwinding.  The machinery is all the same regardless of where a variable is, so I think face-centered fields should support the same things as cell-centered ones.  I haven't verified for myself that it works beyond rank 1 though.

Reply:
Aside from pre-existing issues in the mesh refinement code (which I will fix in the next PR), you should be able to define up to rank 3 face centered quantities. I didn't think through how the reflecting boundary conditions should work in that case though.

I was interested to know if there was some advantage to evolving the stress tensor on  a staggered grid.